### ISSUE TITLE
sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper

### ISSUE THREAD

**Issue by Cactusbone** [dokku/dokku#7159]
### Description of problem

When running commands with dokku in docker, I'm getting "sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper"

See https://github.com/dokku/dokku/issues/7018#issuecomment-2345779873 and https://github.com/dokku/dokku/issues/7018#issuecomment-2359392657

Basically, in `Dockerfile`:
- nginx-vhosts is installed on line 37, with systemctl available, configuring sudoer with systemctl 
- systemctl get removed on line 62

any later calls uses `sv`, but It's not in sudoer file.

### Steps to reproduce

Use dokku in a docker container

### dokku report $APP_NAME

-

**Comment by wfondrie** [comment]
_Reposted from #7018_:

I just ran across this in the v0.34.9 Docker image and I think @Cactusbone is correct. I checked `/etc/sudoers.d/dokku-nginx` within the container and it did indeed reference `systemctl` instead of `sv`. I also updated it to allow `sv` instead using:
```
sed -i "s|/usr/bin/systemctl|$(which sv)|g" /etc/sudoers.d/dokku-nginx
``` 

After this update, things seem to work as expected.

I'm happy to put in a PR to fix the container, but there are few options:
1. Remove `systemctl` earlier in the build (I don't know if this will cause other installation steps to fail, but I would guess not).
2. Add something similar to the above `sed` command to path it in the container - this option seems too fragile to me.
3. Update the logic in `dokku/plugins/nginx-vhosts/install` to use an environment variable explicitly specifying a `systemctl` or `sv` path and falling back to the current logic if it is unset.

Let me know if one of these sounds good, or if there is a better course of action.

I'm can also open a new issue about this, if that is is helpful.

**Comment by josegonzalez** [comment]
This will be fixed in `0.35.x`. Basically has to do with systemd detection. Systemd is installed by default in focal - and actually a ton of packages somehow depend on it in noble - so we need to move away from the fake detection we perform now into a forced-set of `sv` as the init.

**Comment by josegonzalez** [comment]
Here is the fix (for Noble): https://github.com/dokku/dokku/pull/7165/commits/5493ca5978f3d172af91418eeb60fbb02b37de13

Closing as this is confirmed fixed on my end, and should drop shortly.
